crafting-dead-core/src/main/java/com/craftingdead/core/capability/gun/AimableGun.java

```diff
@@ -53,6 +70,40 @@ public float getZoomMultiplier(Entity entity, ItemStack itemStack) {
     return Optional.empty();
   }
 
+  @Override
+  public void toggleRightMouseAction(ILiving<?, ?> living, boolean sendUpdate) {
+    if(canAim) {
+      super.toggleRightMouseAction(living, sendUpdate);
+    }
+  }
+
+  @Override
+  protected void processShoot(ILiving<?, ?> living, ItemStack itemStack) {
+    Entity entity = living.getEntity();
+
+    GunItem gunItem = super.getGunItem();
+
+    super.processShoot(living, itemStack);
+
+    if(this.isAiming(entity, itemStack)
+            && gunItem.hasBoltAction()
+            && super.getAttachments().stream().anyMatch(AttachmentItem::isScope)) {
+
+      toggleRightMouseAction(living, false);
+      canAim = false;
+
+      Timer timer = new Timer();
+      timer.schedule(new TimerTask() {

```

Comment:
Do this without a timer, I suggest using the `lastShotMs` variable and overriding the tick method to check the time passed.